packages/evm-connector/contracts/core/RedstoneConsumerBase.sol

```
@@ -21,6 +21,16 @@ abstract contract RedstoneConsumerBase is CalldataExtractor {
 
   /* ========== VIRTUAL FUNCTIONS (MAY BE OVERRIDDEN IN CHILD CONTRACTS) ========== */
 
+  /**
+   * @dev This function must be implemented by the child consumer contract.
+   * It should return dataServiceId which DataServiceWrapper will use if not provided explicite .
+   * If not overriden, value will always have to be provided explicite in DataServiceWrapper.
+   * @return dataServiceId being consumed by contract
+   */
+  function getDataServiceId() public view virtual returns (string memory) {
+    return "";
```

I would propose to revert here with an error, as we consider "" as invalid value, so maybe we can simply throw the error here

---

Good idea done